### PR TITLE
Remove BlockHeaderHash

### PR DESCRIPTION
**Pull request by Protryon**
Deprecated in favor of `Digest`

### CODE REVIEW

File: network/src/sync/blocks/aggro.rs [hunk 1]
```diff
@@ -20,7 +20,6 @@ use crate::{Cache, Node, Payload, Peer, SyncBase, SyncInbound};
 use anyhow::*;
 use indexmap::IndexSet;
```
Comment (niklaslong) [inline network/src/sync/blocks/aggro.rs:21]: I believe this is an unused import, would be good to remove as a fly-by.

File: network/tests/handshake.rs [hunk 2]
```diff
@@ -185,13 +184,13 @@ async fn reject_non_version_messages_before_handshake() {
...
     let mut peer_stream = TcpStream::connect(node.expect_local_addr()).await.unwrap();
-    let block_hash = BlockHeaderHash::new([0u8; 32].to_vec());
+    let block_hash = Digest::from([0u8; 32]);
     write_message_to_stream(Payload::GetBlocks(vec![block_hash]), &mut peer_stream).await;
     assert_node_rejected_message(&node, &mut peer_stream).await;
 
     // GetSync
     let mut peer_stream = TcpStream::connect(node.expect_local_addr()).await.unwrap();
-    let block_hash = BlockHeaderHash::new([0u8; 32].to_vec());
+    let block_hash = Digest::from([0u8; 32]);
```
Comment (niklaslong) [inline network/tests/handshake.rs:193]: Do we want a constant for the block hash length?

**Review by niklaslong — APPROVED** [review]

File: network/src/sync/blocks/aggro.rs (hunk 1 above)
Comment (Protryon) [reply network/src/sync/blocks/aggro.rs:21]: done in another PR

File: network/tests/handshake.rs (hunk 2 above)
Comment (Protryon) [reply network/tests/handshake.rs:193]: it's a random test, not worried about it.

**Pull request merged**